smart_open/s3.py

```diff
@@ -97,6 +98,8 @@ def open(
     multipart_upload_kwargs: dict, optional
         Additional parameters to pass to boto3's initiate_multipart_upload function.
         For writing only.
+    version_id: str, optional
+        Version of the object. If None, will fetch the most recent version.
```

Comment:
This comment makes sense for when we're reading an object via S3.

How does this parameter behave when writing?

Reply:
test result:
AssertionError: OSError not raised

Need to create a new exception?

Reply:
From boto3's point of view, does it make sense to pass a version number when writing? Or is it something that should **never** happen?